File: src/main/java/pl/ogiba/dwarf/scenes/main/IMainPresenter.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pl.ogiba.dwarf.scenes.main;

/**
 *
 * @author ogiba
 */
public interface IMainPresenter {
    void checkConnection();
    
    void loadData();
    
    void loadCollections();
    
    void loadSelectedCollection(String collection);
    
    void proceedInsertAction();
}

File: src/main/java/pl/ogiba/dwarf/scenes/main/IMainPresenter.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pl.ogiba.dwarf.scenes.main

/**
 *
 * @author ogiba
 */
interface IMainPresenter {
    fun checkConnection()

    fun loadData()

    fun loadCollections()

    fun loadSelectedCollection(collection: String)

    fun proceedInsertAction()
}
